src/app/landing/shared/services/repository.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable, OnDestroy } from "@angular/core";
+import { CoursesByTechnology, CoursesRepository } from "../interfaces/course";
+import { mockCourses } from "./mockCourses";
+import { Observable, Subject, delay, of, retry, shareReplay, takeUntil } from "rxjs";
+import { mockStutendsFeedback } from "./mockStudentsFeedback";
+import { StudentFeedback, StudentFeedbacksRepository } from "../interfaces/student-feedback";
+import { Mentor, MentorsRepository } from "../interfaces/mentor";
+import { mockMentors } from "./mockMentors";
+
+@Injectable({providedIn: 'root'})
+export class RepositoryService implements 
```

Comment:
This service looks too complicated.
Let's discuss it after other reviews will be done.